zipkin-storage/elasticsearch-http/src/main/java/zipkin/storage/elasticsearch/jest/RestClient.java

```diff
@@ -0,0 +1,397 @@
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static com.google.common.util.concurrent.Futures.getUnchecked;
+import static com.google.common.util.concurrent.Futures.transform;
+
+/**
+ * An implementation of {@link InternalElasticsearchClient} that wraps a {@link
+ * io.searchbox.client.JestClient}
+ */
+public final class RestClient extends InternalElasticsearchClient {
+
+  /**
+   * Because the {@link RestClient} is forced to send the index names in the URL, too many indices
+   * (or long names) can quickly exceed the 4096-byte limit commonly imposed on a single HTTP line.
+   * This limit is not entirely within the control of Elasticsearch, either, as it may be imposed by
+   * arbitrary HTTP-speaking middleware along the request's route. Some more details can be found on
+   * this "wontfix" issue: https://github.com/elastic/elasticsearch/issues/7298
+   */
+  private static final int MAX_INDICES = 100;
+
+  public static final class Builder implements InternalElasticsearchClient.Builder {
+    List<String> hosts = Collections.singletonList("http://localhost:9200");
+    boolean flushOnWrites;
+
+    /**
+     * Unnecessary, but we'll check it's not null for consistency's sake.
+     */
+    @Override public Builder cluster(String cluster) {
+      checkNotNull(cluster, "cluster");
+      return this;
+    }
+
+    /**
+     * A comma separated list of elasticsearch hostnodes to connect to, in http://host:port or
+     * https://host:port format. Defaults to "http://localhost:9200".
+     */
+    @Override public Builder hosts(List<String> hosts) {
+      this.hosts = checkNotNull(hosts, "hosts");
+      return this;
+    }
+
+    @Override public Builder flushOnWrites(boolean flushOnWrites) {
+      this.flushOnWrites = flushOnWrites;
+      return this;
+    }
+
+    @Override public Factory buildFactory() {
+      return new Factory(this);
+    }
+
+    public Builder() {
+    }
+  }
+
+  private static final class Factory implements InternalElasticsearchClient.Factory {
+    final List<String> hosts;
+    final boolean flushOnWrites;
+
+    Factory(Builder builder) {
+      this.hosts = ImmutableList.copyOf(builder.hosts);
+      this.flushOnWrites = builder.flushOnWrites;
+    }
+
+    @Override public InternalElasticsearchClient create(String allIndices) {
+      return new RestClient(this, allIndices);
+    }
+  }
+
+  @VisibleForTesting final JestClient client;
+  private final String[] allIndices;
+  private final boolean flushOnWrites;
+
+  private RestClient(Factory f, String allIndices) {
+    JestClientFactory factory = new JestClientFactory();
+    factory.setHttpClientConfig(new HttpClientConfig.Builder(f.hosts)
+        .defaultMaxTotalConnectionPerRoute(6) // matches "regular" TransportClient node conns
+        .maxTotalConnection(6 * 10) // would be 20 otherwise, or ~3 routes
+        .connTimeout(Ints.checkedCast(TimeUnit.MILLISECONDS.convert(10, TimeUnit.SECONDS)))
```

Comment:
I'll change it, but as to the "why" – I would've preferred the java 8 `Duration.ofSeconds(10).toMillis()` (and the checked cast is there because apparently Jest uses ints for time instead of longs), but this is the closest I could come in Java 7.